src/merge.rs

```diff
@@ -0,0 +1,96 @@
+use std::ops::Bound;
+
+// pub trait Add<Rhs = Self> {
+//     /// The resulting type after applying the `+` operator.
+//     #[stable(feature = "rust1", since = "1.0.0")]
+//     type Output;
+
+//     /// Performs the `+` operation.
+//     ///
+//     /// # Example
+//     ///
+//     /// ```
+//     /// assert_eq!(12 + 1, 13);
+//     /// ```
+//     #[must_use]
+//     #[stable(feature = "rust1", since = "1.0.0")]
+//     fn add(self, rhs: Rhs) -> Self::Output;
+// }
+
+/// Merge bounds
+pub(crate) trait BoundMerge<Rhs = Self> {
+    /// Merge start bound
+    fn start_merge(&mut self, rhs: &Rhs);
+    /// Merge end bound
+    fn end_merge(&mut self, rhs: &Rhs);
+}
+
+impl<K> BoundMerge for Bound<K>
+where
+    K: Clone + Ord,
+{
+    fn start_merge(&mut self, rhs: &Self) {
```

Comment:
Please explain the logic in the comment, otherwise it's hard to understand the motivation and logic.

Reply:
Name the function `merge_start`